pkg/graveler/graveler.go

```diff
@@ -1745,6 +1778,24 @@ func (g *KVGraveler) Commit(ctx context.Context, repositoryID RepositoryID, bran
 	return newCommitID, nil
 }
 
+func (g *KVGraveler) retryBranchUpdate(ctx context.Context, repositoryID RepositoryID, branchID BranchID, f BranchUpdateFunc) error {
```

Comment:
1. I think we have more than one backoff/retry packages in the project we can use
2. We should sleep and check context between in order to break if needed and wait between retries

Reply:
done